internal/statesync/syncer.go

```diff
@@ -275,10 +283,17 @@ func (s *syncer) Sync(ctx context.Context, snapshot *snapshot, chunks *chunkQueu
 	// Optimistically build new state, so we don't discover any light client failures at the end.
 	state, err := s.stateProvider.State(pctx, snapshot.Height)
 	if err != nil {
+		if err == context.DeadlineExceeded {
+			return sm.State{}, nil, errDeadlineExceeded
```

Comment:
we wrap the error that we return using the [`%w` format directive](https://blog.golang.org/go1.13-errors#TOC_3.3.). I'm not sure there's a need for this extra case. The caller can use `error.Is(err, Context.DeadlineExceeded)`.